### DISCUSSION TITLE
global using??

### DISCUSSION THREAD

**Discussion by hxhgxy** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3264] [Q&A] [answered]
In some files, I found that no related using, but no error message. for example:

`using System.Windows.Media;

namespace MaterialDesignColors
{
    public interface ISwatch
    {
        string Name { get; }
        IEnumerable<Color> Hues { get; }
        IDictionary<MaterialDesignColor, Color> Lookup { get; }
    }
}`

Codes above are from ISwatch.cs. IEnumerable is from System.Collections.Generic. Why it works without using system.collections.generic?

I tried to search global using in the project, but result is none.

Thanks！

**Comment by Keboo** [comment, marked as answer]
Hi @hxhgxy 

Yes your assumption is correct. This works because of a global using statements. Specifically, those are defined [in the Directory.Build.props file](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/master/Directory.Build.props#L19). You can find more information about that file [here](https://learn.microsoft.com/visualstudio/msbuild/customize-by-directory?view=vs-2022)

**Comment by hxhgxy** [comment]
thank you very much!
